src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -2262,10 +2264,28 @@ public SecurityTokenManager getTokenManager() {
 
     @Override
     public PluginSubject getPluginSubject(Plugin plugin) {
-        Set<String> clusterActions = new HashSet<>();
-        clusterActions.add(BulkAction.NAME);
-        PluginSubject subject = new ContextProvidingPluginSubject(threadPool, settings, plugin);
-        evaluator.updatePluginToClusterActions(subject.getPrincipal().getName(), clusterActions);
+        PluginSubject subject;
+        if (!client && !disabled && !SSLConfig.isSslOnlyMode()) {
+            subject = new ContextProvidingPluginSubject(threadPool, settings, plugin);
+            String pluginPrincipal = subject.getPrincipal().getName();
+            URL resource = plugin.getClass().getClassLoader().getResource("plugin-permissions.yml");
+            RoleV7 pluginPermissions;
+            if (resource == null) {
+                log.warn("plugin-permissions.yml not found on classpath");
```

Comment:
Is the absence of plugin-permissions.yml an expected state or unexpected state? If it is an expected state, I guess we should log max on level info or even just debug. Also, we might want to include the name of the plugin.